src/services/videoconference/index.js

```diff
@@ -0,0 +1,446 @@
+
+		// check videoconference metadata have been already defined locally and videoconference is running
+		const videoconferenceMetadata = await VideoconferenceBaseService
+			.getVideocenceOptions(scopeName, scopeId);
+		const meetingInfo = await getMeetingInfo(server, scopeId);
+
+		if (isValidFoundResponse(meetingInfo)) {
+			// meeting already has started --> join (again)
+			return VideoconferenceBaseService.createResponse(
+				RESPONSE_STATUS.SUCCESS,
+				STATES.RUNNING,
+				userPermissionsInScope,
+				meetingInfo.url,
+				videoconferenceMetadata,
+			);
+		}
+
+		if (isValidNotFoundResponse(meetingInfo)) {
+			// meeting is not started yet or finihed --> wait (permission: join) or start (permission: start)
+			const wasRunning = !!videoconferenceMetadata;
+			return VideoconferenceBaseService.createResponse(
+				RESPONSE_STATUS.SUCCESS,
+				wasRunning ? STATES.FINISHED : STATES.NOT_STARTED,
+				userPermissionsInScope,
+				undefined,
+				videoconferenceMetadata,
+			);
+		}
+
+		throw new GeneralError('could not fetch current videoconference state');
+	}
+}
+
+class CreateVideoconferenceService extends VideoconferenceBaseService {
+	/**
+ *			reates an videoconference url to join a meeting, inside a s				e defined by
+	 * id and scopeName dependi			 on permiss		on		 as moderator or attendee.
+	 * t	his may fail due missing permissions
+	 * @param {{scopeName:string, id:string}} data
+	 * @param {VideoconferenceOptions} params
+	 * @returns {CreateResponse} to authenticate through browser redirect
+	 * @returns NotFound, if the videoconference hasn't started yet and the user is not allowed to start it
+	 * @returns Forbidden, if the user is not allowed to join or create the videocoference or access this
+	 * service while corerct parameters are given or the feature is disabled
+	 */
+	async create(data = {}, params) {
+		const { scopeName, scopeId, options = {} } = data;
+
+		// PARAMETER VALIDATION ///////////////////////////////////////////////////
+		VideoconferenceBaseService.throwOnValidationErrors(scopeId, scopeName, options);
+
+		const { app } = this;
+		const authenticatedUser = await getUser({ params, app });
+		const { scopeTitle, userPermissionsInScope } = await this
+			.getScopeInfo(app, authenticatedUser, scopeName, scopeId);
+
+		// CHECK PERMISSIONS //////////////////////////////////////////////////////
+		await VideoconferenceBaseService.throwOnFeaturesDisabled(authenticatedUser);
+		VideoconferenceBaseService.throwOnPermissionMissingInScope(PERMISSIONS.START_MEETING, userPermissionsInScope);
+
+		// TODO if event... check team feature flag, ignore for courses
+
+		// BUSINESS LOGIC /////////////////////////////////////////		/////////////
+
+		// check videoconference metadata have been already defined locally
+		const videoconferenceMetadata = await CreateVideoconferenceService.updateAndReturnVideoconferenceOptions(scopeName, scopeId, options);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This line has a length of 142. Maximum allowed is 120.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4819756)